synedrion/src/paillier/keys.rs

```diff
@@ -161,17 +202,25 @@ impl<P: PaillierParams> SecretKeyPaillierPrecomputed<P> {
     }
 
     pub fn rns_split(&self, elem: &P::Uint) -> (P::HalfUintMod, P::HalfUintMod) {
-        // TODO (#77): zeroize intermediate values
-
         // May be some speed up potential here since we know p and q are small,
         // but it needs to be supported by `crypto-bigint`.
-        let p_rem = *elem % NonZero::new(self.sk.p.into_wide()).unwrap();
-        let q_rem = *elem % NonZero::new(self.sk.q.into_wide()).unwrap();
-        let p_rem_half = P::HalfUint::try_from_wide(p_rem).unwrap();
-        let q_rem_half = P::HalfUint::try_from_wide(q_rem).unwrap();
+        let mut p_rem = *elem % NonZero::new(self.sk.p.expose_secret().into_wide()).unwrap();
+        let mut q_rem = *elem % NonZero::new(self.sk.q.expose_secret().into_wide()).unwrap();
+        let mut p_rem_half = P::HalfUint::try_from_wide(p_rem).unwrap();
+        let mut q_rem_half = P::HalfUint::try_from_wide(q_rem).unwrap();
 
         let p_rem_mod = p_rem_half.to_mod(self.precomputed_mod_p());
         let q_rem_mod = q_rem_half.to_mod(self.precomputed_mod_q());
+
+        // TODO (#77): zeroize intermediate values
```

Comment:
Perhaps they can be wrapped in `SecretBox` so that we can let RAII do its job? Or would that make too much boilerplate?

Reply:
I think it would be very noisy and difficult to read. 

I wonder if a macro could help us to be able to say, dunno, `secrets_containing_block! { /* lots of secret calculations here, intermediate values included */ }` and write normal code inside the block and be sure that everything inside will be zeroized and return values will be `SecretBox`ed. 

Could be a rather big undertaking though.